### PR TITLE
Page on known issues

### PR DESCRIPTION
**Pull request by konnov**
Added the page on known issues.

- [ ] ~Tests added for any new code~
- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Documentation added for any new functionality
- [x] Entry added to [UNRELEASED.md](./UNRELEASED.md) for any new functionality

### CODE REVIEW

File: UNRELEASED.md [hunk 1]
```diff
@@ -17,3 +17,4 @@
 * Type checker: add support for type aliases,
   e.g., `@typeAlias FOO = [a: Int, b: Int]`, see #704
 * Manual: updated the HOWTO on type annotations with type aliases
+* Manual: the manual page on known issues
```
Comment (shonfeder) [inline UNRELEASED.md:20]:
```suggestion
* Manual: added manual page on known issues
```

File: docs/src/SUMMARY.md [hunk 2]
```diff
@@ -26,6 +26,7 @@
     - [Five minutes of theory](./apalache/theory.md)
 - [TLC Configuration Files](./apalache/tlc-config.md)
 - [The Snowcat Type Checker](./apalache/typechecker-snowcat.md)
+- [Known Issues](./apalache/known-issues.md)
```
Comment (shonfeder) [inline docs/src/SUMMARY.md:29]: I wonder if this should be after "supported features" (at least), but maybe as the very last section?

File: docs/src/apalache/known-issues.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# Known issues
+
+This page collects known issues that were reported by the users.
+
+## Deadlock detection
+
+Deadlock detection is imprecise. It may report false negatives, see [Issue
+711][].
+
+**Affected versions:** 0.11.x, 0.15.x
```
Comment (shonfeder) [inline docs/src/apalache/known-issues.md:10]: Does this mean it the issue was introduced in 0.11? Or do we think it's always been present?

**Review by shonfeder — APPROVED** [review]

File: docs/src/apalache/known-issues.md (hunk 3 above)
Comment (konnov) [reply docs/src/apalache/known-issues.md:10]: I think it was always present, as it has never been our focus.

Comment (shonfeder) [reply docs/src/apalache/known-issues.md:10]: That's what I figured! Thanks :)

Comment (shonfeder) [inline docs/src/apalache/known-issues.md:10]:
```suggestion
**Affected versions:** <= 0.15.x
```

File: docs/src/SUMMARY.md (hunk 2 above)
Comment (konnov) [reply docs/src/SUMMARY.md]: changed the order

**Pull request merged**